### DISCUSSION TITLE
Simplified pages definition

### DISCUSSION THREAD

**Discussion by jyrkidn** [filamentphp/filament#3690] [Ideas]
We have a lot of resources that are just a CRUD, so now we have to always create 4 extra pages per resource.

I know the [simple, modal resources](https://filamentphp.com/docs/2.x/admin/resources/getting-started#simple-modal-resources) exists, but I would like to keep them in separate pages to still be able to use the relation managers.

Is it possible to do something like

```php
public static function getPages(): array
{
    return [
        'index' => BaseListResource::make(static::class)
            ->setActions([
                CreateAction::make(),
            ]),
        'create' => BaseCreateResource::make(static::class),
        'edit' => BaseEditResource::make(static::class)
            ->setActions([
                DeleteAction::make(),
            ]),
    ];
}
```

So there are some base pages with basic functionality that can be extended and if you want more, a page still can be added.

**Comment by danharrin** [comment]
Hey, I don't understand the problem with the extra files though, they get created automatically and you don't touch them?
